### PR TITLE
feat(ee): add `name` field for user model, add api to update it

### PR DESCRIPTION
**Pull request by darknight**
For closing https://github.com/TabbyML/tabby/issues/2054

- add new migration to alter `users` table
  - `name` could be `NULL` for back compatibility
- update `UserDAO` to add `name` field, add `update_user_name` to execute `SQL UPDATE`
- implement new endpoint `update_user_name` for `AuthenticationService`
- implement new mutation endpoint `update_user_name`

- run `make update-db-schema` to update `ee/tabby-db/schema/schema.sql` & `ee/tabby-db/schema/schema.svg`
- run `tabby-schema/examples/update-schema.rs` to update `schema.graphql` file
- udpate `README`

### CODE REVIEW

File: ee/tabby-webserver/src/service/auth.rs [hunk 1]
```diff
@@ -219,6 +219,18 @@ impl AuthenticationService for AuthenticationServiceImpl {
         Ok(self.db.get_user_avatar(id.as_rowid()?).await?)
     }
 
+    async fn update_user_name(&self, id: &ID, name: String) -> Result<()> {
+        if is_demo_mode() {
+            bail!("Changing profile data is disabled in demo mode");
+        }
+        if name.is_empty() || name.len() > 100 {
```
Comment (darknight) [inline ee/tabby-webserver/src/service/auth.rs:226]: Not sure if we should limit name lengh this way.

**Comment by darknight** [comment]
Manual test result:

Screenshot 2024-05-12 161217

Screenshot 2024-05-12 161235

**Comment by darknight** [comment]
code coverage report `Rate limit reached` error, and cargo bloat report `Resource not accessible by integration`. I suspect this is due to external pull request from forked repository, but not very sure about it.

File: ee/tabby-webserver/src/service/auth.rs (hunk 1 above)
Comment (wsxiaoys) [reply ee/tabby-webserver/src/service/auth.rs:226]: 1. please move input verification in `update_user_name` in `schema/mod.rs`.
2. please verify with certain standard rules, e.g no space, no special characters (!%$^&...), proper length.

**Review by wsxiaoys — COMMENTED** [review]
Otherwise LGTM.

As a followup, please fill user.name field when user registered from oauth identity providers.

**Comment by darknight** [comment]
> Otherwise LGTM.
> 
> As a followup, please fill user.name field when user registered from oauth identity providers.

Correct me if I'm wrong, currently when user register from oauth, we only fetch user email info then create user accordingly.
Now we also want to fetch `username` info, and call `update_user_name` during the register process?

One more concern is, in this PR I didn't put `UNIQUE` constraint on the new `name` field, just like `email` or `auth_token`, this is more like product decision, may need your suggestion. cc: @wsxiaoys

File: ee/tabby-schema/src/schema/auth.rs [hunk 2]
```diff
@@ -278,6 +279,26 @@ pub struct PasswordChangeInput {
     pub new_password2: String,
 }
 
+#[derive(Validate)]
+pub struct UpdateUserNameInput {
+    #[validate(length(
+        min = 2,
+        code = "username",
+        message = "Username must be at least 2 characters"
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/auth.rs:287]:
```suggestion
        message = "Name must be at least 2 characters"
```

File: ee/tabby-schema/src/schema/auth.rs [hunk 3]
```diff
@@ -278,6 +279,26 @@ pub struct PasswordChangeInput {
...
+    ))]
+    #[validate(length(
+        max = 20,
+        code = "username",
+        message = "Username must be at most 20 characters"
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/auth.rs:292]:
```suggestion
        message = "Name must be at most 20 characters"
```

File: ee/tabby-schema/src/schema/auth.rs [hunk 4]
```diff
@@ -278,6 +279,26 @@ pub struct PasswordChangeInput {
...
+    ))]
+    #[validate(regex(
+        code = "username",
+        path = "crate::schema::constants::USERNAME_REGEX",
+        message = "Invalid username, only alphabet and space character are allowed, and it must start and end with alphabet"
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/auth.rs:297]:
```suggestion
        message = "Invalid name, only alphabet and space character are allowed, and it must start and end with alphabet"
```

Is non-english characters, e.j german / chinese valid name with this regex?

File: ee/tabby-schema/src/schema/auth.rs [hunk 5]
```diff
@@ -278,6 +279,26 @@ pub struct PasswordChangeInput {
     pub new_password2: String,
 }
 
+#[derive(Validate)]
+pub struct UpdateUserNameInput {
+    #[validate(length(
+        min = 2,
+        code = "username",
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/auth.rs:286]:
```suggestion
        code = "name",
```

File: ee/tabby-schema/src/schema/auth.rs [hunk 6]
```diff
@@ -278,6 +279,26 @@ pub struct PasswordChangeInput {
...
+        message = "Username must be at least 2 characters"
+    ))]
+    #[validate(length(
+        max = 20,
+        code = "username",
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/auth.rs:291]:
```suggestion
        code = "name",
```

File: ee/tabby-schema/src/schema/auth.rs [hunk 7]
```diff
@@ -278,6 +279,26 @@ pub struct PasswordChangeInput {
...
+        message = "Username must be at most 20 characters"
+    ))]
+    #[validate(regex(
+        code = "username",
```
Comment (wsxiaoys) [inline ee/tabby-schema/src/schema/auth.rs:295]:
```suggestion
        code = "name",
```

**Review by wsxiaoys — COMMENTED** [review]
Otherwise LGTM

**Comment by wsxiaoys** [comment]
Please also rebase against main

**Pull request merged**